### PR TITLE
Resolve invalid selector `isEqualToString` being called #781

### PR DESCRIPTION
**Pull request by mitchell-johnson**
Resolve issue where invalid selector `isEqualToString` being called on a number #781

### CODE REVIEW

**Comment by tirodkar** [comment]
https://developer.apple.com/documentation/uikit/uiaccessibilityidentification/1623132-accessibilityidentifier 

Accessibility Identifier is always an NSString, so it has to implement `isEqualToString:`. Can you provide an example of what you're facing so we can see just how this might be able to fix the issue?

Also, you need to sign our CLA.

**Comment by mitchell-johnson** [comment]
I discovered my issue. I had accidentally added an `accessibilityIdentifier` as a BOOL in an element on the affected page. although it's not valid to have a non string `accessibilityIdentifier` there is nothing to stop you doing it accidentally or deliberately. So I wonder if it's worth keeping this check?

**Comment by mitchell-asb** [comment]
Yes I am ok with this PR

File: AppFramework/Matcher/GREYMatchers.m [hunk 1]
```diff
@@ -151,7 +151,11 @@ + (void)initialize {
     if (element.accessibilityIdentifier == accessibilityID) {
       return YES;
     }
-    return [element.accessibilityIdentifier isEqualToString:accessibilityID];
+
```
Comment (wuhao5) [inline AppFramework/Matcher/GREYMatchers.m]: how about `[accessibilityID isEqualToString:element.accessibilityIdentifier]`?

Comment (mitchell-asb) [reply AppFramework/Matcher/GREYMatchers.m]: This is actually a really good idea.

Comment (mitchell-johnson) [reply AppFramework/Matcher/GREYMatchers.m]: This is actually a really good idea.

**Comment by mitchell-johnson** [comment]
@wuhao5 Thats a great idea and adds no extra overhead

**Pull request closed without merging**